gcovr/gcov.py

```diff
@@ -203,15 +204,21 @@ def guess_source_file_name_heuristics(
     if os.path.exists(fname):
         return fname
 
+    # 3. Try using relative path from object dir
+    if obj_dir is not None:
+        fname = os.path.normpath(os.path.join(obj_dir, gcovname))
+        if os.path.exists(fname):
+            return fname
+
     # Get path of gcda file
     source_fname_dir = os.path.dirname(source_fname)
 
-    # 3. Try using the path to the gcda as the source directory
+    # 4. Try using the path to the gcda as the source directory
     fname = os.path.join(source_fname_dir, gcovname)
     if os.path.exists(fname):
         return os.path.normpath(fname)
 
-    # 4. Try using the path to the gcda file as the source directory, removing the path part from the gcov file
+    # 5. Try using the path to the gcda file as the source directory, removing the path part from the gcov file
     fname = os.path.join(source_fname_dir, os.path.basename(gcovname))
     return fname
```

Comment:
One thing I saw while fixing the test. If the file doesn't exist why is this path returned? I was wondering about the output path in the test result.
```suggestion
    # 5. Try using the path to the gcda file as the source directory, removing the path part from the gcov file
    fname = os.path.join(source_fname_dir, os.path.basename(gcovname))
    if os.path.exists(fname):
        return os.path.normpath(fname)

    # return the original path from the gcda file.
    return gcovname
```

Reply:
I also wondered about that part, but didn't want to change anything I could not test.
I am not sure if we should change that in this PR

Reply:
but you are write -> if it does not exist there is no benefit